packages/core/src/forms/control/control.element.ts

```diff
@@ -0,0 +1,292 @@
+
+  @query('.suffix') private suffixAction: HTMLElement;
+
+  @event() layoutChange: EventEmitter<ControlLayout>;
+
+  protected supportsPrefixSuffixActions = true;
+
+  protected observers: (MutationObserver | ResizeObserver)[] = [];
+
+  static get styles() {
+    return [baseStyles, styles];
+  }
+
+  render() {
+    return html`
+      ${this.hiddenLabel ? html`<slot name="label" cds-layout="display:screen-reader-only"></slot>` : ''}
+      <div
+        cds-layout="${this.layout === 'vertical' ? 'vertical gap:sm' : 'horizontal gap:lg'} align:stretch"
+        class="private-host ${this.isRTL ? 'rtl' : ''}"
+      >
+        ${this.primaryLabelTemplate}
+        <div
+          cds-layout="${this.layout === 'compact' ? 'horizontal' : 'vertical'} ${this.controlWidth !== 'compact'
+            ? 'align:horizontal-stretch'
+            : ''} gap:sm"
+        >
+          <div cds-layout="horizontal gap:sm wrap:none">
+            <div
+              cds-layout="horizontal align:top wrap:none ${this.controlWidth === 'compact'
+                ? 'align:shrink'
+                : 'align:horizontal-stretch'}"
+              class="input-container"
+            >
+              ${this.inputTemplate} ${this.prefixTemplate}
+              <slot name="input"></slot>
+              ${this.suffixTemplate}
+            </div>
+            ${this.hiddenLabel ? '' : getStatusIcon(this.status)}
+          </div>
+          ${this.messagesTemplate}
+        </div>
+      </div>
+    `;
+  }
+
+  protected get inputTemplate() {
+    return html``;
+  }
+
+  protected get prefixDefaultTemplate() {
+    return html``;
+  }
+
+  protected get suffixDefaultTemplate() {
+    return html``;
+  }
+
+  private get primaryLabelTemplate() {
+    return html`
+      ${!this.hiddenLabel
+        ? html` <cds-internal-control-label
+            .disabled="${this.disabled}"
+            cds-layout="align:shrink align:top"
+            action="primary"
+          >
+            <slot name="label"></slot>
+          </cds-internal-control-label>`
+        : ''}
+    `;
+  }
+
+  protected get messagesTemplate() {
+    return html`${this.messages?.length
+      ? html`<slot
+          name="message"
+          cds-layout="${this.layout === 'compact' ? 'align:shrink align:vertical-center' : ''}"
+          class="messages"
+        ></slot>`
+      : ''}`;
+  }
+
+  private get prefixTemplate() {
+    return html`
+      <div cds-layout="align:shrink align:vertical-center" class="prefix">
+        <div cds-layout="horizontal gap:xs">
+          ${this.prefixDefaultTemplate}
+          <slot name="prefix"></slot>
+        </div>
+      </div>
+    `;
+  }
+
+  private get suffixTemplate() {
+    return html`
+      <div cds-layout="align:shrink align:vertical-center" class="suffix">
+        <div cds-layout="horizontal gap:xs">
+          <slot name="suffix"></slot>
+          ${this.suffixDefaultTemplate}
+        </div>
+      </div>
+    `;
+  }
```

Comment:
Learned that lit templates can be broken up into getters without some of the perf hits angular has (without onPush change detection)

Reply:
nice!